### ISSUE TITLE
make everything in a cz_ customizable through toml/json and etc

### ISSUE THREAD

**Issue by Atticuszz** [commitizen-tools/commitizen#1278]
### Description

customizing user options is tricky and rigid now.

despite the option from cmd,user configuration should be customized easily from toml,json,etc.

### Possible Solution

refactor customization function in v5

**Comment by bearomorphism** [comment]
Round-2 triage in #1965 lands this as a broad umbrella for "make every aspect of `cz_*` plugins customisable through TOML/JSON". The most concrete sub-task is #1270 (allow partial override of `cz_conventional_commits` via `[tool.commitizen.customize]`), which already has open PR #1570.

Suggest: keep this as the umbrella; close it once #1270/PR #1570 lands and a separate ticket is filed for any remaining piece (custom changelog hooks, configurable `commit_parser` defaults, etc.). Linking from here so the next maintainer pass has the breadcrumbs.
